Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Select.hs

```diff
@@ -62,7 +64,7 @@ handler :: DM.Merchant -> DSelectReq -> DEst.Estimate -> Flow ()
 handler merchant sReq estimate = do
   let merchantId = merchant.id
   searchReq <- QSR.findById estimate.requestId >>= fromMaybeM (SearchRequestNotFound estimate.requestId.getId)
-  farePolicy <- FarePolicyS.findByMerchantIdAndVariant merchant.id estimate.vehicleVariant >>= fromMaybeM NoFarePolicy
+  farePolicy <- getFarePolicyForVariant merchantId (LatLong searchReq.fromLocation.lat searchReq.fromLocation.lon) (LatLong searchReq.toLocation.lat searchReq.toLocation.lon) estimate.vehicleVariant
```

Comment:
let's keep farePolicyId in the estimate/quote table ? that way we would not have to do call location service again and again in each API.
and here you will get the selected vahicle variant so in searchTry you can put the final farePolicyId that is to be used in subsequent APIs.

Reply:
Can I do this as Next PR ?

Reply:
Resolved